support/src/indirect_intrusive_heap.h

```diff
@@ -33,8 +33,10 @@ namespace crimson {
    *
    * heap_info is a data member pointer as to where the heap data in T
    * is stored.
+   *
+   * K is the branching factor of the heap, default is 2 (binary heap).
    */
-  template<typename I, typename T, IndIntruHeapData T::*heap_info, typename C>
+  template<typename I, typename T, IndIntruHeapData T::*heap_info, typename C, uint K = 2>
```

Comment:
Here are the pros and cons of using K as a template argument:
pros:
-- minimal code change (since K is static)

cons:
--  can't adjust the K-value in run-time

I can make necessary changes to load K from a config file if you think using k-way heap is a good idea.